### PR TITLE
RM-73: Remote matching code does not parse Orphanet diseases over the MME

### PR DESCRIPTION
**Pull request by veronikaslc**

### CODE REVIEW

File: core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultPatientToJSONConverter.java [hunk 1]
```diff
@@ -86,17 +86,15 @@ public JSONObject convert(Patient patient, boolean removePrivateData, int includ
         } catch (Exception ex) {
             // Do nothing. These are optional.
         }
+        } catch (Exception ex) {
+            // Do nothing. These are optional.
+        }
```
Comment (buske) [inline core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultPatientToJSONConverter.java]: I don't understand these lines. Does this even compile?

File: core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java [hunk 2]
```diff
@@ -112,14 +112,12 @@ public Patient convert(JSONObject patientJSON)
...
-            Set<Disorder> clinicalDisorders = this.convertDisorders(patientJSON, this.clinicalDisorderTerm,
-                ORDO_VOCABULARY, ApiConfiguration.JSON_DIAGNOSIS);
+            Set<Disorder> disorders = null;
+            Set<Disorder> clinicalDisorders = null;
+            this.convertDisorders(patientJSON, disorders, clinicalDisorders);
             Set<MatchingPatientGene> genes = this.convertGenes(patientJSON);
 
-            if ((features == null || features.isEmpty()) &&
-                (genes == null || genes.isEmpty())) {
+            if ((features == null || features.isEmpty()) && (genes == null || genes.isEmpty())) {
```
Comment (buske) [inline core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java]: Unrelated whitespace change

Comment (veronikaslc) [reply core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java]: better codestyle, && should be either on new line or inline with code

File: core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultPatientToJSONConverter.java (hunk 1 above)
Comment (veronikaslc) [reply core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultPatientToJSONConverter.java]: oops, a glitch, fixed

File: core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java (hunk 2 above)
Comment (buske) [reply core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java]: To make reviewing easier, whitespace/style fixes should be in separate commits from functional changes

Comment (veronikaslc) [reply core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java]: sure, just allowed this small one here :)

**Comment by allasm** [comment]
This still needs to be tested
